shotgun_api3/shotgun.py

```diff
@@ -2206,6 +2333,58 @@ def upload(self, entity_type, entity_id, path, field_name=None, display_name=Non
         attachment_id = int(str(result).split(":")[1].split("\n")[0])
         return attachment_id
 
+    def _get_attachment_upload_info(self, is_thumbnail, filename):
+        """
+        Internal function to get the information needed to upload a file to Cloud storage.
+
+        :param bool is_thumbnail: indicates if the attachment is a thumbnail.
+        :param str filename: name of the file that will be uploaded.
+
+        :returns: dictionary containing the ulpload_info and the url to upload to.
+        :rtype: dict
+        """
+
+        if is_thumbnail:
+            upload_type = 'Thumbnail'
+        else:
+            upload_type = 'Attachment'
+
+        params = {
+            "upload_type" : upload_type,
+            "filename" : filename
+        }
+
+        params.update(self._auth_params())
+
+        upload_url = "/upload/api_get_upload_link_info"
+        url = urlparse.urlunparse((self.config.scheme, self.config.server,
+                                   upload_url, None, None, None))
+
+        # Create opener with extended form post support
+        opener = self._build_opener(FormPostHandler)
```

Comment:
It seems we have this logic in a couple of places:

- https://github.com/shotgunsoftware/python-api/blob/master/shotgun_api3/shotgun.py#L2022
- https://github.com/shotgunsoftware/python-api/blob/master/shotgun_api3/shotgun.py#L2188

at a glance, seems there could be value in addin a helper method that could be used in all three places. (sorry if i am overlooking something!)